test/suite/test_encrypt09.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_encrypt09.py
+#   Test some error conditions with the libsodium encryption extension.
+#
+
+import os, run, random
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+#
+# Test sodium encryption configuration.
+# This exercises the error paths in the encryptor's customize method when
+# used for per-table encryption.
+#
+class test_encrypt09(wttest.WiredTigerTestCase):
+    uri = 'file:test_encrypt09'
+
+    # To test the sodium encryptor, we use secretkey= rather than
+    # setting a keyid, because for a "real" (vs. test-only) encryptor,
+    # keyids require some kind of key server, and (a) setting one up
+    # for testing would be a nuisance and (b) currently the sodium
+    # encryptor doesn't support any anyway.
+    #
+    # It expects secretkey= to provide a hex-encoded 256-bit chacha20 key.
+    # This key will serve for testing purposes.
+    sodium_testkey = '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef'
+
+    # Unlike in test_encrypt08, the nokey case is not an error; if no key is
+    # given no separate encryptor is generated and no error occurs.
+
+    # Note that the twokeys, nothex, and badsize cases do not (currently) get to
+    # the extension at all because (apparently) secretkey= is not allowed for
+    # per-table encryption. If that changes, the proper expected messages are the
+    # same as those in test_encrypt08.
```

Comment:
Please try to rewrite the comments so that they don't refer to other specific test files. Those could get renumbered or moved and no one would know or remember in the future to change this comment to reflect a new name. It is okay to have it stand on its own and refer to "database-wide" versus "per-table".

Reply:
In that case much of that comment can be pruned; it's there to justify why the expected results are different from the other test, but if reference to the other test is not expected, that's moot. I'll update it.